src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/custom.py

```diff
@@ -258,6 +256,9 @@ def upload_block_blob():
             'timeout': timeout
         }
 
+        if tier:
```

Comment:
I would recommend the approach shown below (lines 262-263) since it makes it clear *why* this code is separate. With this approach (which is technically correct if the argument is registered correctly), someone in the future might see it and, in the name of simplification, add it to the above block. That would then likely break older profiles.  @derekbekoe what do you think?